src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -227,10 +234,37 @@ public bool ValidateProcessedBlock(Block processedBlock, TxReceipt[] receipts, B
         {
             if (receipts[i].Error is not null && receipts[i].GasUsed == 0 && receipts[i].Error == "invalid")
             {
-                if (_logger.IsWarn) _logger.Warn($"- invalid transaction {i}");
+                if (_logger.IsWarn) _logger.Error($"- invalid transaction {i}");
                 error = error ?? BlockErrorMessages.InvalidTxInBlock(i);
             }
         }
+
+        if (suggestedBlock.Deposits is not null)
+        {
+            List<Deposit> depositList = [];
+            for (int i = 0; i < processedBlock.Transactions.Length; i++)
+            {
+                foreach (var log in receipts[i].Logs)
+                {
+                    if (log.LoggersAddress == spec.Eip6110ContractAddress)
+                    {
+                        var depositDecoder = new DepositDecoder();
+                        Deposit? deposit = depositDecoder.Decode(new RlpStream(log.Data));
+                        depositList.Add(deposit);
+                    }
+                }
+            }
+
+            Hash256 expectedDepositsRoot = suggestedBlock.Header.DepositsRoot;
+            Hash256 actualDepositsRoot = new DepositTrie(depositList.ToArray()).RootHash;
+            if (actualDepositsRoot != expectedDepositsRoot)
+            {
+                if (_logger.IsWarn) _logger.Error($"- deposits root : expected {expectedDepositsRoot}, got {actualDepositsRoot}");
+                error = error ?? BlockErrorMessages.InvalidDepositsRoot(expectedDepositsRoot, actualDepositsRoot);
+            }
+        }
```

Comment:
I am not sure why this is done here and not when processing the actual block. If deposits root is not calculated there then the `isValid` check  will be incorrect.